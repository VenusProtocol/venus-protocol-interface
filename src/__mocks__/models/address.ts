const address = '0x3d759121234cd36F8124C21aFe1c6852d2bEd848';
export default address;
