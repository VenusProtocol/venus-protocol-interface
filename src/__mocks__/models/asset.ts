import BigNumber from 'bignumber.js';
import { Asset } from 'types';

export const assetData: Asset[] = [
  {
    key: 0,
    id: 'sxp',
    img: '/static/media/sxp.78951004.png',
    vimg: '/static/media/vsxp.b4a90bb0.png',
    symbol: 'SXP',
    decimals: 18,
    tokenAddress: '0x47bead2563dcbf3bf2c9407fea4dc236faba485a',
    vsymbol: 'vSXP',
    vtokenAddress: '0x2ff3d0f6990a40261c66e1ff2017acbc282eb6d0',
    supplyApy: new BigNumber('0.05225450324405023'),
    borrowApy: new BigNumber('-2.3062487835658776'),
    xvsSupplyApy: new BigNumber('0.11720675342484096'),
    xvsBorrowApy: new BigNumber('4.17469243006608279'),
    collateralFactor: new BigNumber('0.5'),
    tokenPrice: new BigNumber('1.2786734'),
    liquidity: new BigNumber('80364658.759955212394069853'),
    borrowCaps: new BigNumber('0'),
    treasuryBalance: new BigNumber('0'),
    walletBalance: new BigNumber('100'),
    supplyBalance: new BigNumber('90'),
    borrowBalance: new BigNumber('0'),
    collateral: false,
    percentOfLimit: '0',
    treasuryTotalBorrowsCents: new BigNumber('70925716.2839193373613181'),
    treasuryTotalSupplyCents: new BigNumber('278311516.880071415163614'),
    treasuryTotalSupply: new BigNumber('19339683254955736'),
    treasuryTotalBorrows: new BigNumber('1852935.597521220541385584'),
    xvsPerDay: new BigNumber('19999999'),
  },
  {
    key: 1,
    id: 'usdc',
    img: '/static/media/usdc.93c65b88.png',
    vimg: '/static/media/vusdc.ea4a1e03.png',
    symbol: 'USDC',
    decimals: 18,
    tokenAddress: '0x8ac76a51cc950d9822d68b83fe1ad97b32cd580d',
    vsymbol: 'vUSDC',
    vtokenAddress: '0xeca88125a5adbe82614ffc12d0db554e2e2867c8',
    supplyApy: new BigNumber('3.887242555711379188'),
    borrowApy: new BigNumber('-5.361233028654066'),
    xvsSupplyApy: new BigNumber('1.353105649796123742'),
    xvsBorrowApy: new BigNumber('1.670327607690572731'),
    collateralFactor: new BigNumber('0.8'),
    tokenPrice: new BigNumber('0.9999364'),
    liquidity: new BigNumber('17029519.59031024692963176'),
    borrowCaps: new BigNumber('0'),
    treasuryBalance: new BigNumber('0'),
    walletBalance: new BigNumber('0'),
    supplyBalance: new BigNumber('100'),
    borrowBalance: new BigNumber('0'),
    collateral: false,
    percentOfLimit: '0',
    treasuryTotalBorrowsCents: new BigNumber('858721657509436.623174884152'),
    treasuryTotalSupplyCents: new BigNumber('1000183891879506332139.9292470565707358'),
    treasuryTotalSupply: new BigNumber('47171999131879271527200010830'),
    treasuryTotalBorrows: new BigNumber('73128318.509651061457900627'),
    xvsPerDay: new BigNumber('19999999'),
  },
  {
    key: 2,
    id: 'usdt',
    img: '/static/media/usdt.ad15d58d.png',
    vimg: '/static/media/vusdt.326a3d1c.png',
    symbol: 'USDT',
    decimals: 18,
    tokenAddress: '0x55d398326f99059ff775485246999027b3197955',
    vsymbol: 'vUSDT',
    vtokenAddress: '0xfd5840cd36d94d7229439859c0112a4185bc0255',
    supplyApy: new BigNumber('3.593608909332766999'),
    borrowApy: new BigNumber('-4.9748661428011145'),
    xvsSupplyApy: new BigNumber('0.421719501189155143'),
    xvsBorrowApy: new BigNumber('0.522209972682294832'),
    collateralFactor: new BigNumber('0.8'),
    tokenPrice: new BigNumber('1.00024602'),
    liquidity: new BigNumber('55341028.864399487015632422'),
    borrowCaps: new BigNumber('0'),
    treasuryBalance: new BigNumber('0'),
    walletBalance: new BigNumber('900'),
    supplyBalance: new BigNumber('0'),
    borrowBalance: new BigNumber('40'),
    collateral: false,
    percentOfLimit: '0',
    treasuryTotalBorrowsCents: new BigNumber('3158444721.332384892254'),
    treasuryTotalSupplyCents: new BigNumber('100980412010115687912488517.7789655772798633'),
    treasuryTotalSupply: new BigNumber('5029972090817266864401527367893625'),
    treasuryTotalBorrows: new BigNumber('232511166.920938849475104194'),
    xvsPerDay: new BigNumber('19999999'),
  },
  {
    key: 3,
    id: 'busd',
    img: '/static/media/busd.e164e45f.png',
    vimg: '/static/media/vbusd.a3f5790b.png',
    symbol: 'BUSD',
    decimals: 18,
    tokenAddress: '0xe9e7cea3dedca5984780bafc599bd69add087d56',
    vsymbol: 'vBUSD',
    vtokenAddress: '0x95c78222b3d6e262426483d42cfa53685a67ab9d',
    supplyApy: new BigNumber('2.886396363044176106'),
    borrowApy: new BigNumber('-4.050271277344538'),
    xvsSupplyApy: new BigNumber('0.678420831753642169'),
    xvsBorrowApy: new BigNumber('0.852697602175970714'),
    collateralFactor: new BigNumber('0.8'),
    tokenPrice: new BigNumber('1.00000922'),
    liquidity: new BigNumber('36544929.350553781829295969'),
    borrowCaps: new BigNumber('0'),
    treasuryBalance: new BigNumber('0'),
    walletBalance: new BigNumber('110'),
    supplyBalance: new BigNumber('0'),
    borrowBalance: new BigNumber('50'),
    collateral: false,
    percentOfLimit: '0',
    treasuryTotalBorrowsCents: new BigNumber('83910350502.1928887132528691'),
    treasuryTotalSupplyCents: new BigNumber('1054707853878.3839597655397544'),
    treasuryTotalSupply: new BigNumber('51881081291203672464'),
    treasuryTotalBorrows: new BigNumber('142662020.229587308931217432'),
    xvsPerDay: new BigNumber('19999999'),
  },
];
