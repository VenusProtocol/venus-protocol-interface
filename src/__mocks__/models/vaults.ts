import BigNumber from 'bignumber.js';

import { Vault } from 'types';

export const vaults: Vault[] = [
  {
    rewardTokenId: 'xvs',
    stakedTokenId: 'vai',
    lockingPeriodMs: 300000,
    dailyEmissionWei: new BigNumber('144000000000000000000'),
    totalStakedWei: new BigNumber('415000000000000000000'),
    stakingAprPercentage: 12665.060240963856,
  },
  {
    rewardTokenId: 'xvs',
    stakedTokenId: 'xvs',
    lockingPeriodMs: 300000,
    dailyEmissionWei: new BigNumber('144000000000000000000'),
    totalStakedWei: new BigNumber('400000000000000000000000000'),
    stakingAprPercentage: 12.92281835063781,
  },
];
