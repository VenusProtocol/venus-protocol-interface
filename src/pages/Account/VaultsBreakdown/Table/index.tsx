/** @jsxImportSource @emotion/react */
import BigNumber from 'bignumber.js';
import { Table, TableProps, TokenIconWithSymbol } from 'components';
import React, { useMemo } from 'react';
import { useTranslation } from 'translation';
import { Vault } from 'types';
import {
  compareBigNumbers,
  compareNumbers,
  convertWeiToTokens,
  formatPercentageToReadableValue,
} from 'utilities';

import { routes } from 'constants/routing';

import { useStyles } from './styles';

export interface VaultTableProps {
  vaults: Vault[];
}

export const VaultTable: React.FC<VaultTableProps> = ({ vaults }) => {
  const { t } = useTranslation();
  const styles = useStyles();

  const tableColumns: TableProps<Vault>['columns'] = useMemo(
    () => [
      {
        key: 'asset',
        label: t('account.vaultsBreakdown.table.column.asset'),
        selectOptionLabel: t('account.vaultsBreakdown.table.column.asset'),
        renderCell: vault => <TokenIconWithSymbol token={vault.stakedToken} />,
      },
      {
        key: 'apr',
        label: t('account.vaultsBreakdown.table.column.apr'),
        selectOptionLabel: t('account.vaultsBreakdown.table.column.apr'),
        renderCell: vault => formatPercentageToReadableValue(vault.stakingAprPercentage),
        sortRows: (rowA, rowB, direction) =>
          compareNumbers(rowA.stakingAprPercentage, rowB.stakingAprPercentage, direction),
      },
      {
        key: 'stake',
        label: t('account.vaultsBreakdown.table.column.stake'),
        selectOptionLabel: t('account.vaultsBreakdown.table.column.stake'),
        renderCell: vault =>
          convertWeiToTokens({
            valueWei: new BigNumber(vault.userStakedWei || 0),
            token: vault.stakedToken,

            returnInReadableFormat: true,
            addSymbol: true,
          }),
        sortRows: (rowA, rowB, direction) =>
          compareBigNumbers(rowA.userStakedWei, rowB.userStakedWei, direction),
      },
    ],
    [vaults],
  );

  return (
    <Table
      css={styles.table}
      title={t('account.vaultsBreakdown.table.title')}
      data={vaults}
      columns={tableColumns}
      rowKeyExtractor={row =>
        `account-vaults-breakdown-table-item-${row.stakedToken.address}-${
          row.rewardToken.address
        }-${row.poolIndex || 0}-${row.lockingPeriodMs || 0}`
      }
      initialOrder={{
        orderBy: tableColumns[2], // Order by stake initially
        orderDirection: 'desc',
      }}
      getRowHref={() => routes.vaults.path}
      breakpoint="xs"
    />
  );
};

export default VaultTable;
