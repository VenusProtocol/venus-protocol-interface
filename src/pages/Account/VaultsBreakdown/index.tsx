import React from 'react';
import { Vault } from 'types';

export interface VaultsBreakdownProps {
  vaults: Vault[];
}

const VaultsBreakdown: React.FC<VaultsBreakdownProps> = () => <></>;

export default VaultsBreakdown;
