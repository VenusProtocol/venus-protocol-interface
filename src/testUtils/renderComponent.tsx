import React from 'react';
import { render } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import { ToastContainer } from 'react-toastify';

import { Web3Wrapper } from 'clients/web3';
import { AuthContext, IAuthContextValue } from 'context/AuthContext';
import { SuccessfulTransactionModalProvider } from 'context/SuccessfulTransactionModalContext';
import Theme from 'theme';
import { RefreshContextProvider } from 'context/RefreshContext';
import { VaiContext, IVaiContextValue } from 'context/VaiContext';
import { MuiThemeProvider } from 'theme/MuiThemeProvider/MuiThemeProvider';
import BigNumber from 'bignumber.js';

const renderComponent = (
  children: React.ReactElement | (() => React.ReactElement),
  {
    authContextValue = {},
    vaiContextValue = {},
  }: {
    authContextValue?: Partial<IAuthContextValue>;
    vaiContextValue?: Partial<IVaiContextValue>;
  } = {},
) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
        cacheTime: 0,
      },
    },
  });
  const defaultAuthContextValues = {
    login: jest.fn(),
    logOut: jest.fn(),
    openAuthModal: jest.fn(),
    closeAuthModal: jest.fn(),
    account: undefined,
    ...authContextValue,
  };
  const defaultVaiContextValues = {
    userVaiMinted: new BigNumber(0),
    userVaiBalance: new BigNumber(0),
    userVaiEnabled: false,
    mintableVai: new BigNumber(0),
    ...vaiContextValue,
  };

  const renderRes = render(
    <Theme>
      <Web3Wrapper>
        <QueryClientProvider client={queryClient}>
          <MuiThemeProvider>
            <AuthContext.Provider value={defaultAuthContextValues}>
              <RefreshContextProvider>
                <VaiContext.Provider value={defaultVaiContextValues}>
                  <SuccessfulTransactionModalProvider>
                    <BrowserRouter>
                      <ToastContainer />

                      <Switch>
                        <Route
                          path="/"
                          component={typeof children === 'function' ? children : () => children}
                        />
                      </Switch>
                    </BrowserRouter>
                  </SuccessfulTransactionModalProvider>
                </VaiContext.Provider>
              </RefreshContextProvider>
            </AuthContext.Provider>
          </MuiThemeProvider>
        </QueryClientProvider>
      </Web3Wrapper>
    </Theme>,
  );

  return {
    ...renderRes,
    queryClient,
  };
};

export default renderComponent;
