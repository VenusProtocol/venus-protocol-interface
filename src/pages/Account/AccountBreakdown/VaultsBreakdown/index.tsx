/** @jsxImportSource @emotion/react */
import React from 'react';
import { useTranslation } from 'translation';
import { Vault } from 'types';

import Section from '../Section';
import Table from './Table';

export interface VaultsBreakdownProps {
  vaults: Vault[];
  className?: string;
}

export const VaultsBreakdown: React.FC<VaultsBreakdownProps> = ({ vaults, className }) => {
  const { t } = useTranslation();

  return (
    <Section className={className} title={t('account.vaultsBreakdown.title')}>
      <Table vaults={vaults} />
    </Section>
  );
};

export default VaultsBreakdown;
