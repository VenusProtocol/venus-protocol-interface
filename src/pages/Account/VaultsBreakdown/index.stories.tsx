import { ComponentMeta } from '@storybook/react';
import React from 'react';

import { vaults } from '__mocks__/models/vaults';
import { withCenterStory, withRouter } from 'stories/decorators';
import { PALETTE } from 'theme/MuiThemeProvider/muiTheme';

import VaultsBreakdown from '.';

export default {
  title: 'Pages/Account/VaultsBreakdown',
  component: VaultsBreakdown,
  decorators: [withRouter, withCenterStory({ width: 1200 })],
  parameters: {
    backgrounds: {
      default: PALETTE.background.default,
    },
  },
} as ComponentMeta<typeof VaultsBreakdown>;

export const Default = () => <VaultsBreakdown vaults={vaults} />;
