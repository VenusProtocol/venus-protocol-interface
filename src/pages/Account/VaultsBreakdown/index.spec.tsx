import React from 'react';

import { vaults } from '__mocks__/models/vaults';
import renderComponent from 'testUtils/renderComponent';

import VaultsBreakdown from '.';

jest.mock('clients/api');

describe('pages/Account/VaultsBreakdown', () => {
  it('renders without crashing', () => {
    renderComponent(<VaultsBreakdown vaults={vaults} />);
  });

  it('displays content correctly', () => {
    const { container } = renderComponent(<VaultsBreakdown vaults={vaults} />);

    expect(container.textContent).toMatchSnapshot();
  });
});
