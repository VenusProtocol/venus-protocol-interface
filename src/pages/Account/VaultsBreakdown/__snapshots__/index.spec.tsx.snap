// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Account/VaultsBreakdown displays content correctly 1`] = `"VaultsStakedAssetAPRStakesorted descendingVAI12665.06%0 VAIXVS12.92%233.00 XVSSort byStakeâ€‹VAIAPR12665.06%Stake0 VAIXVSAPR12.92%Stake233.00 XVS"`;
